addons/sale_project/models/project.py

```diff
@@ -172,7 +172,8 @@ def _compute_sale_order_id(self):
                 sale_order_id = task.sale_line_id.sudo().order_id
             elif task.project_id.sale_order_id:
                 sale_order_id = task.project_id.sale_order_id
-            if task.commercial_partner_id != sale_order_id.partner_id.commercial_partner_id:
+            if task.commercial_partner_id not in (sale_order_id.partner_id.commercial_partner_id +
+                                                  sale_order_id.partner_shipping_id.commercial_partner_id):
```

Comment:
Normally we use the delivery address only in FSM, right? If yes then your fix should be in FSM instead. 🤔

Reply:
@aju-odoo told me that she thinks it's applicable to all services, this is why I've put the fix in sale_project and not in FSM.

Reply:
Hum okay, can you add a test? 🙂 

Thanks in advance!

Reply:
@xavierbol Well turns out you were right, the shipping partner is set as the customer of the task only in the context of FSM. Will close this PR (so I can delete the branch) and make a new PR with this fix in FSM.